### PR TITLE
Backend-Frontend

### PR DESCRIPTION
**Pull request by slvnperron**
I switched the status to `DONE` because it works, but there's still a few enhancements we need to make.

**Note**: You'll need to recreate your database for this to work as the schema changed

### CODE REVIEW

**Comment by slvnperron** [comment]
I saw the conflict but I'll fix it once reviewed

File: src/db.js [hunk 1]
```diff
@@ -57,7 +59,8 @@ module.exports = (knex, botfile) => {
       throw new Error(`Conversation "${conversationId}" not found`)
     }
 
-    await knex('web_messages').insert({
+    const message = {
+      id: uuid.v4(),
       conversationId: conversationId,
       userId: userId,
       full_name: fullName,
```
Comment (DanyFS) [inline src/db.js]: Why user information or not in a separate table? I know that not a problem for the data, but there will be a lot of duplication in the table.

File: src/db.js [hunk 2]
```diff
@@ -67,16 +70,23 @@ module.exports = (knex, botfile) => {
...
   }
 
-  function appendBotMessage(botName, botAvatar, conversationId, { type, text, raw, data }) {
-    return knex('web_messages').insert({
+  async function appendBotMessage(botName, botAvatar, conversationId, { type, text, raw, data }) {
+    const message = {
+      id: uuid.v4(),
       conversationId: conversationId,
       userId: null,
       full_name: botName,
```
Comment (DanyFS) [inline src/db.js]: Same here?

File: src/socket.js [hunk 3]
```diff
@@ -22,6 +27,16 @@ module.exports = async (bp, config) => {
...
+  bp.events.on('guest.web.ping', (data, from, metadata) => {
+    const { visitorId } = metadata
+
+    bp.events.emit('guest.web.pong', {
+      key: 'value',
+      __room: 'visitor:' + visitorId
+    })
+  })
+
   async function outgoingHandler(event, next) {
```
Comment (DanyFS) [inline src/socket.js]: Do you still need this socket test? Can you remove it?

File: src/views/web/index.jsx [hunk 4]
```diff
@@ -47,45 +52,132 @@ export default class Web extends React.Component {
...
+      const currentConversation = Object.assign({}, this.state.currentConversation, {
+        messages: [...this.state.currentConversation.messages, event]
+      })
+
+      this.setState({ currentConversation })
+    })
+
+    this.props.bp.events.on('guest.web.pong', function() {
+      console.log('PONG!!', arguments)
+    })
```
Comment (DanyFS) [inline src/views/web/index.jsx]: Can you remove it? if it's useless

File: src/db.js (hunk 1 above)
Comment (DanyFS) [inline src/db.js]: Why user information are not in a separate table? I know that's not a big deal for the data, but there will be a lot of duplication in the tables.

File: src/db.js (hunk 2 above)
Comment (DanyFS) [inline src/db.js]: Same here?

File: src/socket.js [hunk 5]
```diff
@@ -22,6 +27,16 @@ module.exports = async (bp, config) => {
     ' This middleware should be placed at the end as it swallows events once sent.'
   })
 
+  // TODO Remove this
+  bp.events.on('guest.web.ping', (data, from, metadata) => {
```
Comment (DanyFS) [inline src/socket.js]: Do you still need this socket test? Can you remove it?

File: src/views/web/index.jsx [hunk 6]
```diff
@@ -47,45 +52,132 @@ export default class Web extends React.Component {
...
+      }
+
+      const currentConversation = Object.assign({}, this.state.currentConversation, {
+        messages: [...this.state.currentConversation.messages, event]
+      })
+
+      this.setState({ currentConversation })
+    })
+
+    this.props.bp.events.on('guest.web.pong', function() {
```
Comment (DanyFS) [inline src/views/web/index.jsx]: Can you remove it? if it's useless

File: src/db.js (hunk 1 above)
Comment (slvnperron) [inline src/db.js]: It's for agents and work with HITL. It too complex to have logic that pulls data from different tables (i.e. User vs Bot vs Agent). Does that make sense?

PS: I think storage cost is less of an issue than computing cost these days

**Comment by slvnperron** [comment]
@DanyFS I've fixed your comments except the userId, let me know what you think and I'll fix it in a new PR if you think it's a big issue. Thanks for the review!

**Pull request merged**